great_expectations/expectations/metrics/util.py

```diff
@@ -461,13 +461,28 @@ def get_sqlalchemy_column_metadata(
 
         dialect_name = execution_engine.dialect.name
         if dialect_name == GXSqlDialect.SNOWFLAKE:
-            return [
+            case_insensitive_columns = [
                 # TODO: SmartColumn should know the dialect and do lookups based on that
-                CaseInsensitiveNameDict(column)
+                # WARNING: Do not edit columns in place. It will affect the underlying inspector object.
+                CaseInsensitiveNameDict(column.copy())
                 for column in columns
             ]
+            # Note: Column reflection fallback does not always return types
+            for col in case_insensitive_columns:
+                if "type" in col:
+                    col["type"] = col["type"].compile(dialect=engine.dialect)
 
-        return columns
+            return case_insensitive_columns
+        else:
+            # WARNING: Do not edit columns in place. It will affect the underlying inspector object.
+            columns_copy = [column.copy() for column in columns]
+
+            # Note: Column reflection fallback does not always return types
+            for column in columns_copy:
+                if "type" in column:
+                    column["type"] = column["type"].compile(dialect=engine.dialect)
```

Comment:
confirming my understanding : this is converting the column type (a `str`) to the equivalent sqlalchemy type?